cluster-autoscaler/utils/drain/drain.go

```diff
@@ -287,3 +296,22 @@ func hasSafeToEvictAnnotation(pod *apiv1.Pod) bool {
 func hasNotSafeToEvictAnnotation(pod *apiv1.Pod) bool {
 	return pod.GetAnnotations()[PodSafeToEvictKey] == "false"
 }
+
+// PodShouldBeTerminated check if the pod should have already been remove by the system.
+// Pod has to be terminated, it must have run over its terminationGracePeriod, and also we add an extra time buffer of 'PodDeletionExtraBufferDuration'
+func PodShouldBeTerminated(pod *apiv1.Pod) bool {
```

Comment:
I think a name like `IsPodLongTerminating` would better convey the meaning here, and would be more in line with existing names (e.g. `LongNotStarted`, `LongUnregistered` in `ClusterStateRegistry`).

Same for the comment - I think something along the lines of "IsPodLongTerminating checks if a pod has been terminating for a long time (pod's terminationGracePeriod + an additional const buffer)" would better convey the meaning here.

We can of course still include the reasoning in the comment, ideally mentioning #3407.